File: re-ssim-caterpillar/src/main/java/com/shinonometn/re/ssim/service/caterpillar/common/SchoolDate.java
package com.shinonometn.re.ssim.service.caterpillar.common;

import java.time.DayOfWeek;
import java.util.Objects;

public class SchoolDate {

    private String term;
    private int week;
    private DayOfWeek day;

    public SchoolDate() {
    }

    public SchoolDate(String term, int week, DayOfWeek day) {
        this.term = term;
        this.week = week;
        this.day = day;
    }

    public String getTerm() {
        return term;
    }

    public void setTerm(String term) {
        this.term = term;
    }

    public int getWeek() {
        return week;
    }

    public void setWeek(int week) {
        this.week = week;
    }

    public DayOfWeek getDay() {
        return day;
    }

    public void setDay(DayOfWeek day) {
        this.day = day;
    }

    @Override
    public String toString() {
        return "SchoolDate{" +
                "term='" + term + '\'' +
                ", week=" + week +
                ", day=" + day +
                '}';
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        SchoolDate that = (SchoolDate) o;
        return week == that.week &&
                Objects.equals(term, that.term) &&
                day == that.day;
    }

    @Override
    public int hashCode() {

        return Objects.hash(term, week, day);
    }
}

File: re-ssim-caterpillar/re-ssim-caterpillar-core/src/main/java/com/shinonometn/re/ssim/caterpillar/core/calendar/SchoolDate.kt
package com.shinonometn.re.ssim.caterpillar.core.calendar

import com.fasterxml.jackson.annotation.JsonFormat

import java.time.DayOfWeek
import java.util.Objects

class SchoolDate {

    // Schools have their terms
    var term: String? = null
    var week: Int = 0
    @get:JsonFormat(shape = JsonFormat.Shape.NUMBER)
    var day: DayOfWeek? = null

    constructor() {}

    constructor(term: String, week: Int, day: DayOfWeek) {
        this.term = term
        this.week = week
        this.day = day
    }

    override fun toString(): String {
        return "SchoolDate{" +
                "termName='" + term + '\''.toString() +
                ", week=" + week +
                ", day=" + day +
                '}'.toString()
    }

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other == null || javaClass != other.javaClass) return false
        val that = other as SchoolDate?
        return week == that!!.week &&
                term == that.term &&
                day == that.day
    }

    override fun hashCode(): Int {
        return Objects.hash(term, week, day)
    }
}
